File: tags.py
#!/usr/bin/env python
# -*- coding=utf-8 -*-

import os

os.chdir('./_posts')

tags = []

for x in os.listdir('.'):
    if not x.endswith(".md"):
        continue

    with open(x, 'r') as fp:
        for line in fp.readlines():
            if 'tags:' in line:
                for x in line.split(':')[1].split(' '):
                    tag = x.strip()
                    if not tag or tag in tags:
                        continue
                    tags.append(tag)
                    break

os.chdir('../_my_tags/')

for tag in tags:
    with open(tag.lower() + '.md', 'w') as fp:
        fp.write("---\n" + "slug: " + tag + "\n" + "name: " + tag + "\n---")

File: tags.php
#!/user/bin/env php
<?php

$tags = [];

foreach (glob("_posts/*.md") as $filename) {
    $lines = file_get_contents($filename);
    preg_match("/\ntags: (.*)\n/", $lines, $match);
    if (!isset($match[1])) {
        continue;
    }
    $temp = explode(" ", $match[1]);
    foreach ($temp as $item) {
        $tag = trim($item);
        if (!in_array($tag, $tags)) {
            $tags[] = $tag;
        }
    }
}

foreach (glob("_my_tags/*.md") as $filename) {
    unlink($filename);
}

if (!is_dir('_my_tags')) {
    mkdir('_my_tags', 0700);
}

foreach ($tags as $tag) {
    $tagLow = strtolower($tag);
    $line = sprintf("---\nslug: %s\nname: %s\n---", $tagLow, $tag);
    file_put_contents("_my_tags/" . $tagLow . ".md", $line);
}
